components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -267,33 +263,52 @@ export function CreateWorkspacePage() {
         const rememberedOptions = (user?.additionalData?.workspaceAutostartOptions || []).find(
             (e) => e.cloneURL === cloneURL,
         );
-        setRememberOptions(!!rememberedOptions);
         if (rememberedOptions) {
             // if it's another org, we simply redirect using the same hash and let the reloaded page handle everything again.
             if (rememberedOptions.organizationId !== currentOrg?.id) {
                 const org = organizations.data.find((o) => o.id === rememberedOptions.organizationId);
                 if (org) {
-                    const redirect = `${location.pathname}?org=${encodeURIComponent(rememberedOptions.organizationId)}${
-                        location.hash
-                    }`;
-                    window.location.href = redirect;
+                    let searchParams = `org=${encodeURIComponent(rememberedOptions.organizationId)}`;
+                    // if autostart was disabled (i.e. user was manually changing the contextURL) we need to pass it on
+                    if (autostart === false) {
+                        searchParams += "&autostart=false";
+                    }
```

Comment:
In case the options are stored for a different org, we update the URL to switch to the other org. In that case all state is reset and we need to pass the `autostart` value if it was set to `false` so after the switch the workspace is not started automatically.